austation/code/modules/food_and_drinks/food/kitchen_machinery/cake_printer.dm

```diff
@@ -0,0 +1,190 @@
+			to_chat(user, "<span class='notice'>You pour the cake batter in [src].</span>")
+			I.reagents.remove_reagent(/datum/reagent/consumable/synthetic_cake_batter, reagent_source.amount_per_transfer_from_this)
+			if(fuel > max_fuel)
+				fuel = max_fuel
+			return ..()
+	if(is_type_in_typecache(I, cake_blacklist) || HAS_TRAIT(I, TRAIT_NODROP) || (I.item_flags & (ABSTRACT | DROPDEL)))
+		return ..()
+	item_scanned = I
+	switch(item_scanned.w_class)
+		if(WEIGHT_CLASS_TINY)
+			fuelcost = 15 / efficiency
+		if(WEIGHT_CLASS_SMALL)
+			fuelcost = 30 / efficiency
+		if(WEIGHT_CLASS_NORMAL)
+			fuelcost = 50 / efficiency
+		if(WEIGHT_CLASS_HUGE)
+			fuelcost = 100 / efficiency
+		if(WEIGHT_CLASS_GIGANTIC)
+			fuelcost = 200 / efficiency
+	to_chat(user, "<span class='notice'>You scan [item_scanned] on [src].</span>")
+	return
+
+/obj/machinery/cake_printer/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		. += "<span class='notice'>It has <b>[fuel]</b> fuel left.<span>"
+		if(!item_scanned)
+			. += "<span class='notice'>Alt-click to reset the scanner, no items are scanned.<span>"
+		else
+			. += "<span class='notice'>Alt-click to reset the scanner, current scanned item is [item_scanned].<span>"
+		. += "<span class='notice'>The status display reads: Fuel consumption reduced by <b>[(efficiency*25)-25]</b>%.<br>Machine can hold up to <b>[max_fuel]</b> units of fuel.<br> Speed is increased by <b>[(speed*100)-100]%</b><span>"
+		if(obj_flags & EMAGGED)
+			. += "<span class='warning'>It's status display looks a bit burnt!"
+		. += "<span class='notice'>The machine takes synthetic cake batter as fuel, which is some flour, some milk and a bit of astrotame.<span>"
+
+/obj/machinery/cake_printer/Exited(atom/movable/AM)
+	if(AM == caked_item)
+		finish_cake()
+	..()
+
+/obj/machinery/cake_printer/Destroy()
+	caked_item = null
+	item_scanned = null
+	. = ..()
+
+/obj/machinery/cake_printer/attack_ai(mob/user)
+	return
+
+/obj/machinery/cake_printer/proc/finish_cake()
+	caked_item.desc = "Wait, it's a cake?"
+
+/obj/machinery/cake_printer/attack_hand(mob/user)
+	if(processing)
+		to_chat(user, "<span class='warning'>[src] is printing!</span>")
+		return
+	if(panel_open)
+		to_chat(user, "<span class='warning'>Close the maintenance panel first!</span>")
+		return
+	if(fuel == 0)
+		to_chat(user, "<span class='warning'>No cake batter in the printer!</span>")
+		return
+	if(fuelcost > fuel)
+		to_chat(user, "<span class='warning'>Not enough fuel, add more cake batter!</span>")
+		return
+	if(!item_scanned)
+		to_chat(user, "<span class='warning'>Scan an item first!</span>")
+		return
+	if(!processing)
+		to_chat(user, "<span class='notice'>You start [src]'s printing process.</span>")
+		if(obj_flags & EMAGGED)
+			caked_item = new/obj/item/reagent_containers/food/snacks/synthetic_cake/toxic(src, item_scanned)
+		else
+			caked_item = new/obj/item/reagent_containers/food/snacks/synthetic_cake(src, item_scanned)
```

Comment:
caked_item is already pathed to synthetic_cake. Try to avoid making new/path where you can too.

Reply:
```suggestion
		if(obj_flags & EMAGGED)
			caked_item = /obj/item/reagent_containers/food/snacks/synthetic_cake/toxic
		caked_item = new(src, item_scanned)
```

You probably shouldn't use a new() argument either. just make a separate proc, or set the var from here

Reply:
separate proc still needs to call new(), no point

Reply:
If there's no "new" then the printer will not produce anything at all

Reply:
I know how `new` works